src/m365/spo/commands/tenant/tenant-homesite-add.spec.ts

```diff
@@ -0,0 +1,197 @@
+    assert.strictEqual(command.name, commands.TENANT_HOMESITE_ADD);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('correctly logs command response', async () => {
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://contoso-admin.sharepoint.com/_api/SPHSite/AddHomeSite`) {
+        return homeSites;
+      }
+
+      throw opts.url;
+    });
+
+    await command.action(logger, { options: { url: homeSite, verbose: true } });
+    assert(loggerLogSpy.calledWith(homeSites));
+  });
+
+  it('adds a home site with the specified URL, isInDraftMode, vivaConnectionsDefaultStart, and audiences', async () => {
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://contoso-admin.sharepoint.com/_api/SPHSite/AddHomeSite`) {
+        return homeSiteConfig;
+      }
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        url: homeSite,
+        isInDraftMode: 'true',
+        vivaConnectionsDefaultStart: 'false',
+        audiences: 'af8c0bc8-7b1b-44b4-b087-ffcc8df70d16',
+        order: 2
+      }
+    });
+
+    const expectedData = {
+      "audiences": [
+        "af8c0bc8-7b1b-44b4-b087-ffcc8df70d16"
+      ],
+      "isInDraftMode": "true",
+      "order": 2,
+      "siteUrl": "https://contoso.sharepoint.com/sites/testcomms",
+      "vivaConnectionsDefaultStart": "false"
+    };
+    assert.deepStrictEqual(postStub.lastCall.args[0].data, expectedData);
+  });
+
+  it('fails validation if the url is not a valid SharePoint url', async () => {
+    const actual = await command.validate({
+      options: {
+        url: "test"
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('correctly handles non-integer order', async () => {
+    const result = await command.validate({
+      options: {
+        url: homeSite,
+        order: 'invalid-order'
+      }
+    }, commandInfo);
+    assert.strictEqual(result, 'invalid-order is not a positive integer');
+  });
+
+  it('correctly handles invalid GUIDs in audiences', async () => {
+    const result = await command.validate({
+      options: {
+        url: homeSite,
+        audiences: 'invalid-guid'
+      }
+    }, commandInfo);
+    assert.strictEqual(result, `The following GUIDs are invalid for the option 'ids': invalid-guid.`);
+  });
+
+  it('passes validation with URL', async () => {
+    const actual = await command.validate({
+      options: {
+        url: homeSite
+      }
+    }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation with URL and optional parameters', async () => {
+    const actual = await command.validate({
+      options: {
+        url: homeSite,
+        isInDraftMode: 'true',
+        vivaConnectionsDefaultStart: 'false',
+        audiences: 'af8c0bc8-7b1b-44b4-b087-ffcc8df70d16',
```

Comment:
Let's pass multiple audiences to verify that a comma-separated list is accepted correctly.